### ISSUE TITLE
Filter/reject using the registry version, not only the package version.

### ISSUE THREAD

**Issue by wwwouter** [raineorshine/npm-check-updates#1215]
I like to filter or reject using the registry version.
Right now the `filter` and `reject` functions only have the current package version as parameter.
I would like an extra parameter, with the version ncu plans on updating the package to.
Or if this is not possible, maybe a new function.

I currently have two use cases.

The first is "say no, to dot oh". So if a new version of `x.0.0` is proposed, I want to skip that.

The second is that sometimes the current version of a package still has some problems and I want to wait for the next release. In my case it's `rc-slider`. I want to upgrade to the version that will be released after `10.0.1`. I don't care if it's `10.0.2`, `10.1.0` or `11.0.1`.

So my `reject` function would look like something like this:
```
  reject: (packageName, currentSemVer, proposedSemVer) => {
        if (proposedSemVer[0].minor === '0' && proposedSemVer[0].patch === '0') {
            return true;
        }
        if (packageName === 'rc-slider' && proposedSemVer[0].semver === '10.0.1') {
            return true;
        }

        return false;
    },
```

Is this something that you would like to support? 

If so, I can try to create a PR.

BTW, I tried `rc-slider  8.x || >=10.0.2` in my package.json.
 
But with `upgrade:false`, the result was

```
rc-slider  8.x || >=10.0.2  →  10.x
```

And with `upgrade:true`

```
rc-slider  8.x || >=10.0.2  →  10.0.1
```

- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

---

## Steps to Reproduce

.ncurc:

```js

```

Dependencies:

```json

```

Steps:

**Comment by wwwouter** [comment]
Thanks, I'll try to have a look this week.

Nevermind the "BTW", I misunderstood how npm-check-updates  works.

**Comment by raineorshine** [comment]
Implemented in #1290 as `filterResult`.
